frame/dapps-staking/src/lib.rs

```diff
@@ -238,9 +238,11 @@ impl<Balance: AtLeast32BitUnsigned + Copy> EraStake<Balance> {
 pub struct StakerInfo<Balance: AtLeast32BitUnsigned + Copy> {
     // Size of this list would be limited by a configurable constant
     stakes: Vec<EraStake<Balance>>,
+    // Size of this list would be limited by a configurable constant
+    // pub beneficiaries: Vec<BeneficiaryInfo<AccountId, Balance>>,
```

Comment:
What about this?

Reply:
I ran into some issues adding the AccountId trait into the struct. 
Apparently the Balance trait requires a default trait, but the AccountId trait doesn't implement the trait only in special cases.

Reply:
> Reward destination enum might require some additions perhaps but in general, calling `claim_staker` should esnure to deposit rewards to the correct _beneficiary_ using the new logic.



Hmmmm... Are you saying I should've only edited the `claim_staker` extrinsic?

Reply:
That would have been much simpler solution, yes.

Reply:
I see.  I think I can change the logic based on that assumption.
That'll mean It'll just take in a new argument specifying the reward destination, right?

Reply:
Not what I had in mind tbh :slightly_smiling_face: 

You can check in the call itself what is the reward destination and IF a delegate exists.